front/lib/api/assistant/jit_actions.ts

```diff
@@ -340,6 +334,50 @@ export async function renderConversationForModelJIT({
     }
   }
 
+  if (selected.length > 0) {
+    const { filesAsXML, hasFiles } = await listConversationFiles({
+      conversation,
+    });
+    if (hasFiles) {
+      const randomCallId =
+        "list_conversation_files_" + Math.random().toString(36).substring(7);
+      const functionName = "list_conversation_files";
+
+      const simulatedAgentMessages = [
+        // 1. We add a message from the agent, asking to use the files listing function
+        {
+          role: "assistant",
+          content: "Please list the files available to this conversation.",
+          function_calls: [
+            {
+              id: randomCallId,
+              name: functionName,
+              arguments: "{}",
+            },
+          ],
+        } as AssistantFunctionCallMessageTypeModel,
+
+        // 2. We add a message with the resulting files listing
+        {
+          function_call_id: randomCallId,
+          role: "function",
+          name: functionName,
+          content: filesAsXML,
+        } as FunctionMessageTypeModel,
+      ];
+
+      // Update the list of messages and the tokens count.
+      for (const m of simulatedAgentMessages) {
+        const approxSimulatedAgentMessagesTokenCount =
+          getTextRepresentationFromMessages([m]).join("").length / 3;
```

Comment:
I don't get it, why do we divide by 3 here ?

Reply:
That how we approximate higher up in the method for the truncate message (3 tokens per characters).
I also hesistated to call the core api to count tokens for real but I didn't know if it was overkill.

Reply:
Ah I see, thanks 👍

Reply:
In upcoming commit, I reworked so I append the simulated messages before calculating token count, this got removed.